src/import/FSHTank.ts

```diff
@@ -419,82 +419,8 @@ export class FSHTank implements Fishable {
-            // - for each index, tell me what the last value is. that's the actual value.
-            // - if one of them has The Value I Seek, then the overall condition is true
-            // - otherwise, the overall condition is false
-            const extensionUrlRules = result.rules.filter(rule => {
-              return (
-                rule instanceof CaretValueRule &&
-                rule.path === '' &&
-                /^extension(\[\d+\])?\.url$/.test(rule.caretPath)
-              );
-            }) as CaretValueRule[];
-            const extensionUrls = new Map<number, string>();
-            extensionUrlRules.forEach(urlRule => {
-              const match = urlRule.caretPath.match(/^extension(\[(\d+)\])?\.url$/);
-              if (match?.[2]) {
-                extensionUrls.set(parseInt(match[2]), urlRule.value.toString());
-              } else {
-                extensionUrls.set(0, urlRule.value.toString());
-              }
-            });
-            for (const [idx, url] of extensionUrls.entries()) {
-              if (url === TYPE_CHARACTERISTICS_EXTENSION) {
-                const caretPaths = [`extension[${idx}].valueCode`];
-                if (idx === 0) {
-                  caretPaths.push('extension.valueCode');
-                }
-                const valueRule = result.rules
-                  .filter(rule => {
-                    return (
-                      rule instanceof CaretValueRule &&
-                      rule.path === '' &&
-                      caretPaths.includes(rule.caretPath)
-                    );
-                  })
-                  .pop() as CaretValueRule;
-                if (
-                  valueRule &&
-                  valueRule.value instanceof FshCode &&
-                  valueRule.value.code === 'can-be-target'
-                ) {
-                  meta.canBeTarget = true;
-                  break;
-                }
-              } else if (url === LOGICAL_TARGET_EXTENSION) {
-                const caretPaths = [`extension[${idx}].valueBoolean`];
-                if (idx === 0) {
-                  caretPaths.push('extension.valueBoolean');
-                }
-                const valueRule = result.rules
-                  .filter(rule => {
-                    return (
-                      rule instanceof CaretValueRule &&
-                      rule.path === '' &&
-                      caretPaths.includes(rule.caretPath)
-                    );
-                  })
-                  .pop() as CaretValueRule;
-                if (valueRule?.value === true) {
-                  meta.canBeTarget = true;
-                  break;
-                }
-              }
-            }
-          }
+          meta.canBeTarget = hasLogicalCharacteristic(result, 'can-be-target');
+          meta.canBind = hasLogicalCharacteristic(result, 'can-bind');
```

Comment:
I refactored all this code in the branch I'm currently working on. This will be a fun rebase for me! (But seriously, don't worry about it).